rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -339,57 +98,41 @@ def _query(
             accounts: Sequence[BTCAddress],
             options: dict[str, Any] | None = None,
     ) -> dict[BTCAddress, FVal] | dict[BTCAddress, tuple[bool, FVal]] | tuple[int, list[BitcoinTx]]:  # noqa: E501
-        """Queries bitcoin explorer APIs, if one fails the next API is tried.
-        The errors from all the queries are included in the resulting remote error if all fail.
-
-        * blockchain.info: Supports all actions
-        * blockcypher: Currently only support for transactions is implemented.
-          TODO: add support for querying blockcypher for the other actions as well.
-        * blockstream.info & mempool.space: Similar APIs. Support for transactions is not
-          implemented for these because they don't handle p2pk properly.
+        """Queries explorer APIs for the specified action as defined in `self.api_callbacks`.
+        If one query fails the next API for that action is tried, and the errors from all the
+        queries are included in the resulting remote error if all fail.
 
         May raise:
         - RemoteError if the queries to all the APIs fail.
         """
+        if (callbacks := self.api_callbacks.get(action)) is None:
+            raise RemoteError(f'No API callbacks found for {self.blockchain!s} {action.name.lower()} query')  # noqa: E501
+
         errors: dict[str, str] = {}
         kwargs: dict[str, Any] = {'accounts': accounts}
-        callbacks = [('blockchain.info', getattr(self, f'_query_blockchain_info_{action.name.lower()}'))]  # noqa: E501
         if action == BtcQueryAction.TRANSACTIONS:
-            kwargs['options'] = options
-            callbacks.append(('blockcypher.com', self._query_blockcypher_transactions))
-        else:
-            callbacks.extend([
-                ('blockstream.info', lambda **kwargs:
-                    getattr(self, f'_query_blockstream_or_mempool_{action.name.lower()}')(
-                        base_url=BLOCKSTREAM_BASE_URL,
-                        **kwargs,
-                    ),
-                 ),
-                ('mempool.space', lambda **kwargs:
-                    getattr(self, f'_query_blockstream_or_mempool_{action.name.lower()}')(
-                        base_url=MEMPOOL_SPACE_BASE_URL,
-                        **kwargs,
-                    ),
-                 ),
-            ])
-
-        for api_name, callback in callbacks:
+            kwargs['options'] = options  # Only transaction queries use extra options
+
+        for callback in callbacks:
+            callback_kwargs = kwargs if callback.kwargs is None else kwargs | callback.kwargs
             try:
-                return callback(**kwargs)
+                callback_fn = getattr(self, callback.function_name.format(action.name.lower()))
```

Comment:
this is what I mention that can fail in a bad way. I would just do callback.query_balances